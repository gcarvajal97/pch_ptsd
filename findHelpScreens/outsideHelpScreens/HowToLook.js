import React, { Component } from 'react';
import { View, StyleSheet, Text} from 'react-native';
import { Linking } from "expo";
import { translate } from '../../components/translateService';

export default class HowToLook extends Component {

  render() {
    return (
      <View>
        <Text style={styles.paragraph}>{translate('howToLook.paragraph1')}</Text>
        <Text style={styles.paragraph}>{translate('howToLook.paragraph2')}</Text>
        <Text style={styles.paragraph}>{translate('howToLook.paragraph3')}</Text>
        <Text style={styles.paragraph}>
        {"\ "}
        <Text style={styles.link} onPress={()=> Linking.openURL('http://www.nctsnet.org/nccts/nav.do?pid=ctr_aud_prnt_gethelp')}
          accessible
          accessibilityLabel={translate('howToLook.accessibilityLabel')}
          accessibilityHint={translate('howToLook.accessibilityHint')}>
          {translate('howToLook.linkName')}
          </Text>
          {"\ "}{translate('howToLook.paragraph4')}
        </Text>
    </View>
    );
  }
}

const styles = StyleSheet.create({
  paragraphTitle: {
    fontWeight: 'bold',
    fontSize: 20,
    alignSelf: 'stretch',
    textAlign: 'left',
    marginTop: 6,
    marginBottom: 3,
    color: '#2089DC'
  },
  paragraph: {
    textAlign: 'left',
    fontSize: 14,
    marginVertical: 2,
  },
  paragraphBold: {
    textAlign: 'left',
    fontSize: 14,
    fontWeight: 'bold',
    marginVertical: 2,
    marginLeft: 10
  },
  link: {
    fontSize: 14,
    color:'#2089DC',
    textDecorationLine: 'underline',
  }
});