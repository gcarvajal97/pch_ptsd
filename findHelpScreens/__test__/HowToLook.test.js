import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from 'react';
import HowToLook from '../outsideHelpScreens/HowToLook';
import renderer from 'react-test-renderer';
import {Linking} from 'expo';

configure({ adapter: new Adapter() });
jest.mock('expo', () => ({
    Linking: {
        openURL: jest.fn()
    }
}));


describe("HowToLook", () => {
    afterEach(() => {
        jest.clearAllMocks();
      });

    it("renders correctly", () => {
        expect(renderer.create(<HowToLook/>).toJSON()).toMatchSnapshot();
    });

    it("clicks external link, opens correct website", () => {
        const wrapper = shallow(<HowToLook/>);
        wrapper.findWhere((node) => node.prop('testID') === 'hyperlink').props().onPress();
        expect(Linking.openURL).toHaveBeenCalledWith('http://www.nctsnet.org/nccts/nav.do?pid=ctr_aud_prnt_gethelp');
    })
})