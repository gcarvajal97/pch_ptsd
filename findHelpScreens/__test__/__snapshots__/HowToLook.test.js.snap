// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToLook renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "fontFamily": "avenir-medium",
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    There are very good and helpful treatments for children who continue to have traumatic stress symptoms.
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "avenir-medium",
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    If you are wondering how to find a child therapist, a good place to start is to talk with your child's doctor or a counselor at your child's school. They can help you figure out the best way of helping your child and your family, and may be able to suggest a counselor or therapist experienced in helping children with traumatic stress.
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "avenir-medium",
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    When looking for a counselor or therapist to help your child with traumatic stress symptoms, it's a good idea to ask questions such as: Do you have experience working with children after trauma? What is your approach to working with this type of problem? What is your success rate? How do you work with parents?
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "avenir-medium",
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
     
    <Text
      accessibilityHint="Navigates to a website on mental health professional information"
      accessibilityLabel="Click here for mental health professional information"
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "color": "#0077C8",
          "fontFamily": "avenir-medium",
          "fontSize": 14,
          "textDecorationLine": "underline",
        }
      }
      testID="hyperlink"
    >
      The National Child Traumatic Stress Network
    </Text>
     
    has useful tips and information on how to find a mental health professional to help your child and family after trauma.
  </Text>
</View>
`;
